### PR TITLE
spring starter can now use all sdk autoconfig properties

### PR DESCRIPTION
**Pull request by zeitlinger**
Also 

- annotation dependency is included in starter now
- explain advanced config with the example of "exclude actuator endpoints from tracing"
- explain resource providers

### CODE REVIEW

File: content/en/docs/languages/java/automatic/spring-boot.md [hunk 1]
```diff
@@ -172,60 +179,217 @@ Set the value to `true` to disable the starter, e.g. for testing purposes.
...
+
+`application.properties`:
+
+```properties
+otel.propagators=tracecontext,b3
+otel.resource.attributes=environment=dev,xyz=foo
+```
 
-#### Tracer Properties
+This style is convenient to use in environment variables:
```
Comment (theletterf) [inline content/en/docs/languages/java/automatic/spring-boot.md:202]:
```suggestion
Use this style in environment variables:
```

File: content/en/docs/languages/java/automatic/spring-boot.md [hunk 2]
```diff
@@ -172,60 +179,217 @@ Set the value to `true` to disable the starter, e.g. for testing purposes.
...
-Finally, the resource attributes can be specified as a comma-separated list, as
-described in the
-[specification](/docs/languages/sdk-configuration/general/#otel_resource_attributes):
+#### Programmatic configuration
 
-```shell
-export OTEL_RESOURCE_ATTRIBUTES="key1=value1,key2=value2"
+You can use the `AutoConfigurationCustomizerProvider` for programmatic
+configuration. Programmatic configuration is recommended for advanced use cases,
+which are not configurable via properties.
```
Comment (theletterf) [inline content/en/docs/languages/java/automatic/spring-boot.md:238]:
```suggestion
which are not configurable using properties.
```

**Review by theletterf — APPROVED** [review]
Just a couple nits. Thanks!

**Comment by jeanbisutti** [comment]
@theletterf Could you please have another look?

**Comment by theletterf** [comment]
Approved!

**Review by cartermp — APPROVED** [review]

**Comment by trask** [comment]
I'm a little late, but this is great 🎉

**Comment by zeitlinger** [comment]
> I'm a little late, but this is great 🎉

Never too late for that :smile:

**Pull request merged**